clue.js

```diff
@@ -0,0 +1,181 @@
+  professorPlum,
+  missScarlet,
+  mrsWhite
+]
+
+const weapon = [
+  rope,
+  gun,
+  knife,
+  wrench,
+  candlestick,
+  pipe
+]
+
+const room = [
+  'hall',
+  'lounge',
+  'study',
+  'library',
+  'kitchen',
+  'ballroom'
+]
+
+// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
+// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
+const randomSelector = array => {
+  return array[Math.floor(Math.random() * array.length)]
+}
+
+// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
+// With a killer, a weapon and a room.
+// The values will be set later.
+let crime = {
+ killer: [],
+ weapon: [],
+ room: ''
+}
+
+// This function will be invoked when you click on the killer card.
+const pickKiller = () => {
+  // This will randomly select a killer from the suspects. And add that to the mystery object.
+  mystery.killer = randomSelector(suspects)
+
+  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
+  document.getElementById('killerCard').style.background = mystery.killer.color;
+  document.getElementById('killerCard').addEventListener('click', pickKiller)
+  document.getElementById('killerName').innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`;
+  document.getElementById('killerAge').innerHTML = `age: ${mystery.killer.age}`;
+  document.getElementById('killerOccupation').innerHTML = `occupation: ${mystery.killer.occupation}`;
+  document.getElementById('killerDescription').innerHTML = `${mystery.killer.description}`;
+  document.getElementById('killerImage').src = mystery.killer.image;
+}
+
+// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+const pickWeapon = () => {
+  mystery.weapon = randomSelector(weapon)
+  document.getElementById('weaponCard').style.background = mystery.killer.color;
+  document.getElementById('weaponCard').addEventListener('click', pickWeapon)
+  document.getElementById('weaponName').innerHTML = `${mystery.weapon.name}`
+  document.getElementById('weaponWeight').innerHTML = `${mystery.weapon.weight}`
+  document.getElementById('weaponImage').src = mystery.weapon.image;
+}
+
+const pickRoom = () => {
+  mystery.room = randomSelector(room)
+  document.getElementById('roomCard').style.background = mystery.killer.color;
+  document.getElementById('roomName').innerHTML = `${mystery.room}`
+}
+
+
+
+// STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
+// 'The murder was committed by Jacob Green, in the living room with a rope.'
+
+function revealMystery() {
+document.getElementById("revealMystery").innerHTML = `The murder was committed by ${mystery.killer.firstName} ${mystery.killer.lastName}, in the ${mystery.room} with a ${mystery.weapon.name}`;
+}
+
+document.getElementById('killerCard').onclick = pickKiller;
```

Comment:
I also called each function at the bottom of my code, so this is great, so they're all grouped together. But I think because you have the `document.getElementById('killerCard').addEventListener('click', pickKiller)` inside each of the functions, this would do this for you if you had them written outside of the function. Maybe something the teacher can confirm as I'm 100% sure!